controllers/util/mw_util.go

```diff
@@ -502,11 +502,28 @@ func (mwu *MWUtil) createOrUpdateManifestWork(
 	}
 
 	if !reflect.DeepEqual(foundMW.Spec, mw.Spec) {
-		mw.Spec.DeepCopyInto(&foundMW.Spec)
-
 		mwu.Log.Info("ManifestWork exists.", "name", mw.Name, "namespace", foundMW.Namespace)
 
-		return mwu.Client.Update(mwu.Ctx, foundMW)
+		for {
+			mw.Spec.DeepCopyInto(&foundMW.Spec)
+
+			err = mwu.Client.Update(mwu.Ctx, foundMW)
+			if err == nil {
+				break
+			}
+
+			if !errors.IsConflict(err) {
+				return err
+			}
+
+			err = mwu.Client.Get(mwu.Ctx,
+				types.NamespacedName{Name: mw.Name, Namespace: managedClusternamespace},
+				foundMW)
+
+			if err != nil {
```

Comment:
What is the purpose of the Get operation here?

Also, if the error we got was a conflict err, should we not retry the update after getting it?

Reply:
Inside the RetryOnConflict function we need to do 3 steps:
1. Reload manifest work resource
2. Update it's status (deepcopy)
3. Update the manifest work resource

Get function can't have conflicts, if we failed in it - no need to proceed, it's real error
Update function return value is inspected by RetryOnConflict function itself and the decision if exit or proceed is taken by it.

Reply:
In my opinion, implementing this pattern is good for the ManifestWork. Because it can be modified by an external entity at runtime and we have no control over it.